File: wlr/OGUR/OGUR/HUD/DeltasHud.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Microsoft.Xna.Framework;
using Microsoft.Xna.Framework.Graphics;
using OGUR.Creatures;
using OGUR.Items;
using OGUR.Management;
using OGUR.Text;
using SPX.Util;

namespace OGUR.HUD
{
    public class DeltasHud:IHud
    {
        private Equipment m_equipment;

        public DeltasHud(ICreature owner, Equipment equipment): base(owner, XnaManager.WindowWidth / 2, XnaManager.WindowHeight / 2)
        {
            m_equipment = equipment;
        }

        public void Draw()
        {
            if (m_isVisible)
            {
                m_textHandler.Draw();
            }
        }

        private GenericItem GetEquipmentIn(int slot)
        {
            if (m_equipment.GetItems().ContainsKey(slot))
            {
                return m_equipment.GetItems()[slot];
            }
            return null;
        }

        private const string delim = "|";
        private const string positive = "+";
        private const string title = "Deltas";
        private string display = "EMPTY";

        public void Update(GenericItem item,bool refresh)
        {
            if (m_isVisible)
            {
                m_textHandler.Update();
                m_textHandler.Clear();
                if (item != null && refresh)
                {
                    if (GetEquipmentIn(Equipment.ClassToSlot(item.GetItemClass())) != null)
                    {
                        StringSquisher.Clear();
                        foreach (var stat in GetEquipmentIn(Equipment.ClassToSlot(item.GetItemClass())).Modifers.GetDeltas(item.Modifers))
                        {
                            StringSquisher.Squish(((stat > 0) ? positive : String.Empty),StringStorage.Get(stat),delim);
                        }
                        display = StringSquisher.Flush();
                    }
                }
                m_textHandler.WriteDefault(title, 30, 260, GetHudOrigin());
                m_textHandler.WriteDefault(display, 30, 290, GetHudOrigin());
            }
        }
    }
}

File: src/aigilas/hud/DeltasHud.java
package aigilas.hud;

import aigilas.creatures.BaseCreature;
import aigilas.items.Equipment;
import aigilas.items.GenericItem;
import aigilas.items.ItemSlot;
import sps.graphics.Renderer;
import sps.text.Text;
import sps.text.TextPool;
import sps.util.StringSquisher;
import sps.util.StringStorage;

public class DeltasHud extends BaseHud {
    private final Equipment _equipment;

    public DeltasHud(BaseCreature owner, Equipment equipment) {
        super(owner, Renderer.get().VirtualWidth / 2, Renderer.get().VirtualHeight / 2);
        _equipment = equipment;
    }

    @Override
    public void draw() {
    }

    private GenericItem getEquipmentIn(ItemSlot slot) {
        if (_equipment.getItems().containsKey(slot)) {
            return _equipment.getItems().get(slot);
        }
        return null;
    }

    private static final String delim = "|";
    private static final String positive = "+";
    private static final String title = "Deltas";
    private String display = "EMPTY";

    private Text heading;
    private Text deltas;

    public void update(GenericItem item, boolean refresh) {
        if (_isVisible) {
            if (item != null && refresh) {
                if (getEquipmentIn(item.getItemClass().Slot) != null) {
                    StringSquisher.clear();
                    for (Integer stat : getEquipmentIn(item.getItemClass().Slot).Modifiers.getDeltas(item.Modifiers)) {
                        StringSquisher.squish(((stat > 0) ? positive : ""), StringStorage.get(stat), delim);
                    }
                    display = StringSquisher.flush();
                }
            }
            if (heading == null || !heading.isVisible() || refresh) {
                if (deltas != null) {
                    heading.hide();
                    deltas.hide();
                }
                heading = TextPool.get().write(title, getInventoryAnchor().add(30, (int) (_dimensions.Y * .2)));
                deltas = TextPool.get().write(display, getInventoryAnchor().add(30, (int) (_dimensions.Y * .1) + 20));
            }
        }
        else {
            if (heading != null) {
                heading.hide();
                deltas.hide();
            }
        }
    }
}
